### PR TITLE
Test and fix zapback

### PR DESCRIPTION
**Pull request by kevingranade**
As mentioned [here](https://discourse.cataclysmdda.org/t/shockers-zap-you-if-bow-conducts-electricity/15586?u=kevin.granade), zapback was triggering in an obscure case where a ranged weapon also had a reach attack.
While investigating I also discovered that said weapons had no way of triggering the reach attack, so I added automatic inserion of an appropriate firing mode to items of this kind on creation and fixed fire action handling.

### CODE REVIEW

File: tests/mondefense_test.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    standard_npc attacker( "Attacker" );
+    item rock( "rock" );
+    attacker.wield( rock );
+    test_zapback( attacker, false );
+}
+
+TEST_CASE( "zapback_npc_nonconductive_unarmed_weapon", "[mondefense]" )
+{
+    standard_npc attacker( "Attacker" );
+    // AFAICT this is the only nonconductive unarmed weapon.
```
Comment (Night-Pryanik) [inline tests/mondefense_test.cpp]: Cestus is another example.

Comment (Night-Pryanik) [inline tests/mondefense_test.cpp]: Cestus is another example.

**Comment by AMurkin** [comment]
`"Striking the tab shocks you!"`

Where NPC hp were set before db742a6?

**Comment by kevingranade** [comment]
This is a separate npc constructor used for making testing npcs.  I want actually able to pin down where it happens normally, for players it happens in newcharacter.cpp

**Pull request merged**